### PR TITLE
syncpwm UI updates

### PR DESCRIPTION
**Pull request by mlyle**
This has seen cursory but not systematic test.  The problems I saw were fixed.  Also fixed was a uavo overflow.

### CODE REVIEW

**Comment by mlyle** [comment]
This supersedes #1642

**Comment by DTFUHF** [comment]
Much better.

The buffer overflow - I thought the number of elements field in a UAVO enum was meant to describe the number of elements in that enum. Is this the case? Or is this fix used to fix a separate bug.

**Comment by mlyle** [comment]
@DTFUHF -- thanks for the initial refactor.  I see why things were the way they were-- the UI code for this output actuator stuff is very fiddly and hard to make do the right thing.  Half of my change is hack, unfortunately. :/

Re: buffer overflow-- It's not the number of possible values for the enum, but the number of instances of that thing in an array.  I missed it in code review and just happened to get lucky to see it scroll by on compile output.

(I think I'm planning to eliminate GCS warnings and try and make them fatal so that it's harder to do something like this).

File: ground/gcs/src/plugins/config/configoutputwidget.cpp [hunk 1]
```diff
@@ -455,15 +465,78 @@ void ConfigOutputWidget::refreshWidgetsValues(UAVObject * obj)
...
+                //
+                QComboBox* ccmb = rateList.at(i);
+                quint32 timerFreq = timerStringToFreq(ccmb->currentText());
+
+                QComboBox* res = resList.at(i);
+                quint32 timerRes = res->currentIndex();
+
+                double maxPulseWidth = 0, uavoMaxPulseWidth = 0;
+
+                // Saturate at the UAVO's maximum value
```
Comment (peabody124) [inline ground/gcs/src/plugins/config/configoutputwidget.cpp]: This comment/variable is inaccurate I think. The UAVO max is in microseconds. However, you are actually calculating the maximum number of microseconds on the physical timer, assuming 16 bits (some aren't, but a reasonable conservative assumption).

So it is really timerMaxPeriodUs

File: ground/gcs/src/plugins/config/configoutputwidget.cpp [hunk 2]
```diff
@@ -455,15 +465,78 @@ void ConfigOutputWidget::refreshWidgetsValues(UAVObject * obj)
...
+                uavoMaxPulseWidth = 65535;
+
+                if (timerRes == ActuatorSettings::TIMERPWMRESOLUTION_12MHZ)
+                    uavoMaxPulseWidth = uavoMaxPulseWidth / 12;
+
+                if (timerFreq != 0)
+                {
+                    maxPulseWidth = 1000000 / timerFreq;
```
Comment (peabody124) [inline ground/gcs/src/plugins/config/configoutputwidget.cpp]: and if we are changing names :D could this be timerPeriodUs

File: ground/gcs/src/plugins/config/outputchannelform.cpp [hunk 3]
```diff
@@ -86,14 +86,6 @@ OutputChannelForm::OutputChannelForm(const int index, QWidget *parent, const boo
     ui.actuatorLink->setChecked(false);
     connect(ui.actuatorLink, SIGNAL(toggled(bool)), this, SLOT(linkToggled(bool)));
 
-    // Trigger when autopilot is connected
-    connect(this, SIGNAL(autoPilotConnected()), this, SLOT(onAutopilotConnect()));
```
Comment (peabody124) [inline ground/gcs/src/plugins/config/outputchannelform.cpp]: why was this removed? isn't it calling a method on the parent that refreshes the initial values (e.g. neutral?)

Comment (mlyle) [inline ground/gcs/src/plugins/config/outputchannelform.cpp]: No.  The parent already gets an autopilot connect event and has it connected.  We were using that to calculate minimum/maximum values, but now the parent does it when it processes the bank.  We don't need to know down here.

**Pull request merged**